src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java

```diff
@@ -63,6 +63,15 @@
     @Argument(fullName = "maxReadsPerAlignmentStart", shortName = "maxReadsPerAlignmentStart", doc = "Maximum number of reads to retain per alignment start position. Reads above this threshold will be downsampled. Set to 0 to disable.", optional = true)
     protected int maxReadsPerAlignmentStart = defaultMaxReadsPerAlignmentStart();
 
+    @Argument(fullName = "alignmentStartStride", shortName = "alignmentStartStride", doc = "Length in bases to include in a single unit of positional downsampling.", optional = true)
```

Comment:
The number of arguments for `AssemblyRegionWalker` is increasing exponentially, and this is a pity for other usages outside `HaplotypeCaller` and `Mutect2` if they aren't need it. Maybe it's time to add an abstract `AssemblyRegionArgumentCollection` ( #argumentCollection to allow hiding some arguments (related with #2371 too, because a `ShardingArgumentCollection` will be useful too).

Like that, instead of a bunch of abstract default getters, only an `AssemblyRegionArgumentCollection` getter will be needed with a default implementation returned by `HaplotypeCaller` and `Mutect2`. I can open a PR with this idea and/or implement it in #2371 (either before or after this PR). What do you think, @droazen?